javascriptv3/example_code/s3/actions/create-bucket.js

```diff
@@ -1,30 +1,64 @@
 // Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
 
-import { fileURLToPath } from "url";
-
 // snippet-start:[s3.JavaScript.buckets.createBucketV3]
-import { CreateBucketCommand, S3Client } from "@aws-sdk/client-s3";
-
-const client = new S3Client({});
+import {
+  BucketAlreadyExists,
+  BucketAlreadyOwnedByYou,
+  CreateBucketCommand,
+  S3Client,
+  waitUntilBucketExists,
+} from "@aws-sdk/client-s3";
 
-export const main = async () => {
-  const command = new CreateBucketCommand({
-    // The name of the bucket. Bucket names are unique and have several other constraints.
-    // See https://docs.aws.amazon.com/AmazonS3/latest/userguide/bucketnamingrules.html
-    Bucket: "bucket-name",
-  });
+/**
+ * Create an Amazon S3 bucket.
+ * @param {{ bucketName: string }} config
+ */
+export const main = async ({ bucketName }) => {
+  const client = new S3Client({});
 
   try {
-    const { Location } = await client.send(command);
+    const { Location } = await client.send(
+      new CreateBucketCommand({
+        // The name of the bucket. Bucket names are unique and have several other constraints.
+        // See https://docs.aws.amazon.com/AmazonS3/latest/userguide/bucketnamingrules.html
+        Bucket: bucketName,
+      }),
+    );
+    await waitUntilBucketExists({ client }, { Bucket: bucketName });
     console.log(`Bucket created with location ${Location}`);
-  } catch (err) {
-    console.error(err);
+  } catch (caught) {
+    if (caught instanceof BucketAlreadyExists) {
```

Comment:
or `BucketAlreadyExists` just in case user doesn't own the bucket but it's nevertheless taken.